### PR TITLE
Improve tests, bats tests should not include setup (or state)

### PR DESCRIPTION
**Pull request by rfay**
Up to now, the bats test were dependent on each other, but we need to make them just dependent on their own setup. This moves the basic build and install into the test script.

### CODE REVIEW

File: package_drupal_script.sh [hunk 1]
```diff
@@ -144,11 +144,11 @@ cp ${REPO_DIR}/example.gitignore ${STAGING_DIR}/sprint/drupal8/.gitignore
 
 echo "Running composer install --quiet"
 composer install --quiet
+popd >/dev/null
 
 # Copy licenses and COPYING notice.
-cp -r ${REPO_DIR}/licenses "$STAGING_DIR/"
-cp ${REPO_DIR}/COPYING "$STAGING_DIR/"
-popd >/dev/null
```
Comment (mradcliffe) [inline package_drupal_script.sh:151]: Whoops. Maybe a good test case to assert the package contents?

File: tests/sane_testbot.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+echo "sanetestbot.sh: Check to see if test machine has what it needs"
```
Comment (mradcliffe) [inline tests/sane_testbot.sh]: Small typo here. Not a big deal.

File: tests/sane_testbot.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+# choco install -y jq 7zip composer zip (gd and curl must be enabled in /c/tools/php72/php.ini)
+# apt-get install jq p7zip-full
+# git clone git://github.com/bats-core/bats-core; cd bats-core; git checkout v1.1.0; sudo ./install.sh /usr/local
+
+for item in curl jq 7z composer perl bats; do
+    command -v $item >/dev/null || ( echo "$item is not installed" && exit 2 )
+done
+docker run --rm -t -v "/$PWD:/junk" busybox ls //junk >/dev/null || ( echo "docker is not running" && exit 3 )
+
+echo "Testbot appears to be sane"
```
Comment (mradcliffe) [inline tests/sane_testbot.sh]: Testbots can be sane? ;-)

**Review by mradcliffe — COMMENTED** [review]
:+1: on moving into a tests directory.

It might be worth it to add a small test to assert the licenses directory since that was a bug fix, right?

Edit: sorry, saw some emails come through and linked directly to the changeset without noticing the label.

File: tests/sane_testbot.sh (hunk 2 above)
Comment (rfay) [reply tests/sane_testbot.sh]: Fixed.

File: package_drupal_script.sh (hunk 1 above)
Comment (rfay) [reply package_drupal_script.sh:151]: I added assertion on these items. Will keep thinking about the rest of the contents, but they'd all cause other failures so it doesn't seem as bad.

**Comment by rfay** [comment]
I think this is ready, would love it if you could take a look @mradcliffe. Wow, did that take up the day. It wasn't supposed to be like that.

File: sprint/start_clean.sh [hunk 4]
```diff
@@ -44,11 +44,8 @@ done
 ddev config --docroot drupal8 --projectname sprint-[ts] --projecttype drupal8
 
 ddev start
-ddev exec git fetch
-ddev exec git reset --hard origin/8.7.x
-ddev exec composer install
-ddev exec drush si standard --account-pass=admin --db-url=mysql://db:db@db:3306/db --site-name="Drupal Sprinting"
-ddev exec drush cr
+ddev exec bash -c 'git fetch && git reset --hard origin/8.7.x && composer install && drush si standard --account-pass=admin --db-url=mysql://db:db@db/db --site-name="Drupal Sprinting" && drush cr'
```
Comment (mradcliffe) [inline sprint/start_clean.sh:48]: #77 would probably be important since this is long-running, but not a blocker for this PR.

**Review by mradcliffe — COMMENTED** [review]
I'm downloading the build artifacts to confirm, and then I think this PR is good.

File: sprint/start_clean.sh (hunk 4 above)
Comment (rfay) [reply sprint/start_clean.sh:48]: Shoot, thought that was in. This particular line actually generates enough output that people will understand that things are going on. But I think an explanatory line would be good here.

**Review by mradcliffe — APPROVED** [review]

**Comment by mradcliffe** [comment]
I probably won't have time to focus on any PRs until late in the day today.

**Pull request merged**